src/pages/workspaces/workspace/Analyses.js

```diff
@@ -477,14 +477,25 @@ const Analyses = _.flow(
           workspace,
           onDismiss: () => setExportingAnalysisName(undefined)
         }),
-        deletingAnalysisName && h(AnalysisDeleter, {
-          printName: getDisplayName(deletingAnalysisName), googleProject, bucketName,
-          toolLabel: getTool(deletingAnalysisName),
-          onDismiss: () => setDeletingAnalysisName(undefined),
-          onSuccess: () => {
-            setDeletingAnalysisName(undefined)
-            refreshAnalyses()
-          }
+        deletingAnalysisName && h(DeleteConfirmationModal, {
+          title: getTool(deletingAnalysisName) ? `Delete ${getTool(deletingAnalysisName)} analysis` : 'Delete analysis',
+          confirmationPrompt: 'Delete analysis',
+          buttonText: 'Delete analysis',
+          objectName: getDisplayName(deletingAnalysisName),
+          onConfirm: async () => {
+            try {
+              await Ajax().Buckets.analysis(
+                googleProject,
+                bucketName,
+                getDisplayName(deletingAnalysisName),
+                getTool(deletingAnalysisName)
+              ).delete()
+              refreshAnalyses()
```

Comment:
There is no need to call `setDeletingAnalysisName(undefined)` in the `onConfirm` case? How does the deletingAnalysisName get cleared?

Reply:
PromptedConfirmationModal calls `onDismiss` after `onConfirm`.

https://github.com/DataBiosphere/terra-ui/blob/f1ad9fbaafb4c8f6513bb5fe1485895679ab868f/src/components/common.js#L564-L571